### DISCUSSION TITLE
Can the records of rabbitmq's monitor queue be persisted?

### DISCUSSION THREAD

**Discussion by ponponon** [rabbitmq/rabbitmq-server#9494] [Questions] [answered]
Requirement: Monitoring can be done after rabbtimq reboot and the monitoring logs are still there.

![图片](https://github.com/rabbitmq/rabbitmq-server/assets/38725104/b21dfd7a-0f73-4f97-b234-f0c536fe3253)

Status: after rabbtimq restart, the previous record is gone, that is, the record is not persistent

**Comment by mkuratczyk** [comment]
No, this data is not stored persistently inside RabbitMQ. Use an external monitoring system like Prometheus to store the metrics.

**Comment by michaelklishin** [comment, marked as answer]
And the docs [explain the specific reasons why we recommend external monitoring](https://www.rabbitmq.com/monitoring.html#with-prometheus).
